File: src/main/java/no/novari/flyt/configuration/model/metadata/InstanceObjectCollectionMetadata.java
package no.novari.flyt.configuration.model.metadata;

import lombok.Builder;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;

@Getter
@Builder
@Jacksonized
public class InstanceObjectCollectionMetadata {
    private final String key;
    private final InstanceMetadataContent objectMetadata;
}

File: src/main/kotlin/no/novari/flyt/configuration/model/metadata/InstanceObjectCollectionMetadata.kt
package no.novari.flyt.configuration.model.metadata

data class InstanceObjectCollectionMetadata(
    var displayName: String? = null,
    var key: String? = null,
    @Suppress("unused")
    var objectMetadata: InstanceMetadataContent? = null,
) {
    companion object {
        @JvmStatic
        fun builder(): Builder = Builder()
    }

    class Builder {
        private var displayName: String? = null
        private var key: String? = null
        private var objectMetadata: InstanceMetadataContent? = null

        fun displayName(displayName: String?) = apply { this.displayName = displayName }

        fun key(key: String?) = apply { this.key = key }

        fun objectMetadata(objectMetadata: InstanceMetadataContent?) = apply { this.objectMetadata = objectMetadata }

        fun build(): InstanceObjectCollectionMetadata =
            InstanceObjectCollectionMetadata(displayName = displayName, key = key, objectMetadata = objectMetadata)
    }
}
